testing/gimmeproj/main.go

```diff
@@ -142,7 +145,21 @@ Administrative commands:
 		fmt.Fprintln(os.Stderr, usage.Error())
 		return nil
 	case "lease":
-		return lease(ctx, flag.Arg(1))
+		// When leasing, keep trying until we reach our configured timeout
+		ctx, cancel := context.WithTimeout(ctx, *waitTime)
+		defer cancel()
+		for ctx.Err() == nil {
+			err := lease(ctx, flag.Arg(1))
+			if err == nil {
+				return err
+			} else if errors.Is(err, ErrNoProjects) {
+				log.Printf("Temporary error: %v\n", err)
```

Comment:
suggestion: It would be good to provide some guidance on what the wait might be, such as "trying every 30 seconds until 30 minutes".